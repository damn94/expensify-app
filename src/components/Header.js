import React from 'react';
import {NavLink, Link} from 'react-router-dom'


//HEADER COMPONENT
const Header = () => (
<div>
		<h1>Expensify</h1>
	{/*button links using NavLink*/}
	<button>
		<NavLink to='/' activeClassName='is-active' exact>Home</NavLink>
	</button>
	<button>
		<NavLink to='/create' activeClassName='is-active'>Add Expense</NavLink>
	</button>
{/*	<button>
		<NavLink to='/edit' activeClassName='is-active'>Edit Expense</NavLink>
	</button>*/}
	<button>
		<NavLink to='/help' activeClassName='is-active'>Help</NavLink>
	</button>
</div>
)


export default Header;