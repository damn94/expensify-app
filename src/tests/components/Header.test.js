
//REPLACE REACTSHALLOWRENDER BY ENZYME
// import ReactShallowRenderer from 'react-test-renderer/shallow';

//need toJson to extract junk  lines in our shallow() function 
// import toJson from 'enzyme-to-json'

import Header from '../../components/Header'
import { shallow } from 'enzyme'
import React from 'react';

test('Should render Header correctly',()=>{

	//create variable for return shallow Func
	const  wrapper = shallow(<Header />)

	//MatchSnapshot with to Json
	// expect(toJson(wrapper)).toMatchSnapshot()

	expect(wrapper).toMatchSnapshot()

//little wthat enzyme can do find selectors
//expect(wrapper.find('h1').text()).toBe('Expensify')
//expect(wrapper.find('button').text()).toBe(3)

})







// //snap shot test ReactShallowRender
// test('should render Header correctly',()=>{
// const renderer = new ReactShallowRenderer;
// renderer.render(<Header />);

// expect(renderer.getRenderOutput()).toMatchSnapshot();
// })
