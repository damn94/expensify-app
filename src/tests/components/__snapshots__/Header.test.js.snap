// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Header correctly 1`] = `
<div>
  <h1>
    Expensify
  </h1>
  <button>
    <NavLink
      activeClassName="is-active"
      exact={true}
      to="/"
    >
      Home
    </NavLink>
  </button>
  <button>
    <NavLink
      activeClassName="is-active"
      to="/create"
    >
      Add Expense
    </NavLink>
  </button>
  <button>
    <NavLink
      activeClassName="is-active"
      to="/help"
    >
      Help
    </NavLink>
  </button>
</div>
`;
